anap1003/assignment1/Q2.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package assignment1;
+
+import java.util.Iterator;
+import java.util.LinkedList;
+
+/**
+ *
+ * @author Ana
+ */
+public class Q2 {
+
+    private static int q2(int k, LinkedList list) {
+
+        int retValue = Integer.MIN_VALUE;
+
+        if (k < 0) {
+            return retValue;
+        }
+
+        if (k + 1 > list.size()) {

```

Comment:
calling list.size() here removes the benefits of using the itrKstepsAhead below as if you are getting the size anyway, then why not use it. Try refactoring so as to not use .size() at all.